src/main/webapp/script.js

```diff
@@ -0,0 +1,37 @@
+function countUnits(){
+  const inputElems = document.getElementsByName("search-units");
+  const arr = []
+  for (let i=0; i<inputElems.length; i++) {       
+    if (inputElems[i].checked){
+      arr.push(inputElems[i].value);
+    }
+  }
+  return arr;
+}
+
+function showClasses() {
+  const classes = document.getElementById("search-results");
+  const className = document.getElementById("search-class").value;
+  const profName = document.getElementById("search-prof").value;
+  const units = countUnits();
+  classes.innerHTML = "";
+  const url = new URL("/search", window.location.origin);
+  url.searchParams.set("className", className);
+  url.searchParams.set("profName", profName);
+  url.searchParams.set("units", units);
+  fetch(url)
+  .then(response => response.json())
+  .then((response) => {
+    return JSON.parse(response.classNames);
+  })
+  .then((classNames) => {
```

Comment:
It doesn't look like just a list of class names is returned? Correct me if I'm missing something though

Reply:
In my servlet, I generate a list of class names and set them to the classNames attribute of my json response

Reply:
There are more things being returned but just as proof of correctness, I am looking at classNames

Reply:
Oooh, now I'm seeing that you're only parsing response.classNames, missed that earlier